azurerm/resource_arm_virtual_network_gateway_connection.go

```diff
@@ -194,9 +305,26 @@ func resourceArmVirtualNetworkGatewayConnectionRead(d *schema.ResourceData, meta
 		d.Set("local_network_gateway_id", conn.LocalNetworkGateway2.ID)
 	}
 
-	d.Set("enable_bgp", conn.EnableBgp)
-	d.Set("routing_weight", conn.RoutingWeight)
-	d.Set("shared_key", conn.SharedKey)
+	if conn.EnableBgp != nil {
+		d.Set("enable_bgp", conn.EnableBgp)
```

Comment:
do we want to dereference this field?

Reply:
This is currently done in ResourceData.Set